test/controllers/propertyDetails/DateCouncilRegisteredControllerSpec.scala

```diff
@@ -103,6 +106,44 @@ class DateCouncilRegisteredControllerSpec extends PlaySpec with GuiceOneServerPe
       val result = dateCouncilRegisteredKnownController.view("1").apply(SessionBuilder.buildRequestWithSession(userId))
       test(result)
     }
+
+    def saveWithAuthorisedUser(inputJson: JsValue, test: Future[Result] => Any): Any = {
+
+      val userId = s"user-${UUID.randomUUID}"
+      val authMock = authResultDefault(AffinityGroup.Organisation, defaultEnrolmentSet)
+      noDelegationModelAuthMocks(authMock)
+      when(mockServiceInfoService.getPartial(ArgumentMatchers.any(), ArgumentMatchers.any(), ArgumentMatchers.any()))
+        .thenReturn(Future.successful(HtmlFormat.empty))
+      when(mockDataCacheConnector.fetchAtedRefData[String](ArgumentMatchers.eq(AtedConstants.DelegatedClientAtedRefNumber))
+        (ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some("XN1200000100001")))
+      when(mockDataCacheConnector.saveFormData[DateCouncilRegistered](ArgumentMatchers.any(), ArgumentMatchers.any())
+        (ArgumentMatchers.any(), ArgumentMatchers.any()))
+        .thenReturn(Future.successful(null))
+      when(mockDataCacheConnector.fetchAndGetFormData[DateCouncilRegisteredKnown](ArgumentMatchers.eq(AtedConstants.NewBuildCouncilRegisteredDateKnown))
+        (ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some(DateCouncilRegisteredKnown(Some(true)))))
+      when(mockDataCacheConnector.fetchAndGetFormData[DateFirstOccupied](ArgumentMatchers.eq(NewBuildFirstOccupiedDate))
+        (ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some(DateFirstOccupied(Some(LocalDate.now())))))
+      when(mockDataCacheConnector.fetchAndGetFormData[DateCouncilRegistered](ArgumentMatchers.eq(NewBuildCouncilRegisteredDate))
+        (ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some(DateCouncilRegistered(Some(LocalDate.now())))))
+      when(mockPropertyDetailsService.saveDraftPropertyDetailsNewBuildDates(ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any(), ArgumentMatchers.any())) thenReturn {
+        Future.successful(1)
+      }
+
+
+
+      when(mockDataCacheConnector.fetchAndGetFormData[DateFirstOccupiedKnown](ArgumentMatchers.eq(NewBuildFirstOccupiedDateKnown))
+        (ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some(DateFirstOccupiedKnown(Some(true)))))
+
+      when(mockBackLinkCacheConnector.saveBackLink(ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any()))
+        .thenReturn(Future.successful(None))
+
+      /*when(mockPropertyDetailsService.retrieveDraftPropertyDetails(ArgumentMatchers.any())(ArgumentMatchers.any(), ArgumentMatchers.any())) thenReturn {
```

Comment:
I'm guessing that this (and subsequent commented out detail) can be removed if commented out and not required?

Reply:
Removed - resolved